diff --git a/dist/index.mjs b/dist/index.mjs
index 87806e0daebe0c457f7efb5fb00a182f95c109a5..e60892d3865f6f9034230bda712a6881a180489b 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -778,15 +778,24 @@ const createImportMagicComments = (options) => {
 const componentsPluginTemplate = {
   filename: "components.plugin.mjs",
   getContents({ options }) {
-    return `import { defineNuxtPlugin } from '#app/nuxt'
-import { lazyGlobalComponents } from '#components'
+    const globalComponents = options.getComponents().filter(c => c.global === true)
+
+    return `import { defineAsyncComponent } from 'vue'
+import { defineNuxtPlugin } from '#app/nuxt'
+
+const components = ${genObjectFromRawEntries(globalComponents.map((c) => {
+  const exp = c.export === 'default' ? 'c.default || c' : `c['${c.export}']`
+  const comment = createImportMagicComments(c)
+
+  return [c.pascalName, `defineAsyncComponent(${genDynamicImport(c.filePath, { comment })}.then(c => ${exp}))`]
+}))}
 
 export default defineNuxtPlugin({
   name: 'nuxt:global-components',` + (options.getComponents().filter((c) => c.global).length ? `
   setup (nuxtApp) {
-    for (const name in lazyGlobalComponents) {
-      nuxtApp.vueApp.component(name, lazyGlobalComponents[name])
-      nuxtApp.vueApp.component('Lazy' + name, lazyGlobalComponents[name])
+    for (const name in components) {
+      nuxtApp.vueApp.component(name, components[name])
+      nuxtApp.vueApp.component('Lazy' + name, components[name])
     }
   }` : "") + `
 })
@@ -819,7 +828,6 @@ const componentsTemplate = {
     return [
       ...imports,
       ...components,
-      `export const lazyGlobalComponents = ${genObjectFromRawEntries(options.getComponents().filter((c) => c.global).map((c) => [c.pascalName, `Lazy${c.pascalName}`]))}`,
       `export const componentNames = ${JSON.stringify(options.getComponents().filter((c) => !c.island).map((c) => c.pascalName))}`
     ].join("\n");
   }
